File: 1twoSum/twoSum.java
class Solution {
    public int[] twoSum(int[] nums, int target) {
        final var dictionary = new HashMap<Integer, Integer>();
        final var iter = Arrays.stream(nums).boxed().collect(Collectors.toList()).listIterator();
        while (iter.hasNext())
        {
            final var index = iter.nextIndex();
            final var num = iter.next();
            if (dictionary.containsKey(num)) return new int[]{index, dictionary.get(num)};
            else dictionary.putIfAbsent(target - num, index);
        }
        return new int[0];
    }
}

File: 1twoSum/twoSum.js
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    const dictionary = new Map
    for (const [index, num] of nums.entries())
        if (dictionary.has(num)) return [index, dictionary.get(num)]
        else dictionary.set(target - num, index)
};
